lib/reservation.rb

```diff
@@ -0,0 +1,26 @@
+require 'date'
+
+module Hotel
+  class Reservation
+
+    attr_reader :room_number, :checkin_date, :checkout_date, :cost
+
+      def initialize(room_number, checkin_date, checkout_date, cost: 200)
+        # check_date_range(checkin_date, checkout_date)
+        @room_number = room_number
+        @checkin_date = checkin_date
+        if checkout_date < checkin_date
+          raise ArgumentError, "check out cannot happen before check in"
```

Comment:
Nice!
Because the nature of raising Errors means that code will halt and stop the program if it reaches it, I might consider refactoring this so it can be a little slimmer...
```ruby
if checkout_date < checkin_date
  raise ArgumentError, "check out cannot happen before check in"
end
```

Mostly, I think you don't need the `else` bit!